pkg/kubelet/kuberuntime/kuberuntime_image.go

```diff
@@ -47,36 +51,27 @@ func (m *kubeGenericRuntimeManager) PullImage(image kubecontainer.ImageSpec, pul
 		return "", err
 	}
 
-	var imageServices []internalapi.ImageManagerService
+	var imageService internalapi.ImageManagerService
 
-	// Get the image service from the map cache for this image
-	imageWithoutTag := strings.Split(image.Image, ":")[0]
-	if is := m.getPodImageService(imageWithoutTag); is != nil {
-		klog.Infof("Got desired image service for image %v, %v", image, is)
-		imageServices = []internalapi.ImageManagerService{is}
+	if is, err := m.GetImageServiceByPod(image.Pod); err != nil {
+		klog.Errorf("GetAllImageServices Failed: %v", err)
 	} else {
-		klog.Infof("Cannot find desired image service for image %s, loop through all image services to pull", image.Image)
-		imageServices, err = m.GetAllImageServices()
-		if err != nil {
-			klog.Errorf("GetAllImageServices Failed: %v", err)
-			return "", err
-		}
+		klog.Infof("Got desired image service for image %v, %v", image, is)
+		imageService = is
 	}
 
 	imgSpec := &runtimeapi.ImageSpec{Image: img}
 	creds, withCredentials := keyring.Lookup(repoToPull)
 	if !withCredentials {
 		klog.V(3).Infof("Pulling image %q without credentials", img)
 
-		for idx, imageService := range imageServices {
-			klog.Infof("Pulling image with image service %d:%v", idx, imageService)
-			imageRef, err := imageService.PullImage(imgSpec, nil, podSandboxConfig)
-			if err != nil {
-				klog.Errorf("Pull image %q failed: %v", img, err)
-			} else {
-				klog.Infof("Got image ref for image %q: %v", image, imageRef)
-				return imageRef, nil
-			}
+		klog.Infof("Pulling image with image service %v", imageService)
+		imageRef, err := imageService.PullImage(imgSpec, nil, podSandboxConfig)
+		if err != nil {
+			klog.Errorf("Pull image %q failed: %v", img, err)
+		} else {
+			klog.Infof("Pull image ref for image %q: %v", image, imageRef)
```

Comment:
nit: for successful info, we may want to lower the logging priority (confusing; higher level like v5+) to reduce the log file size.

Reply:
yes, will do. thanks

Reply:
fixed.